### ISSUE TITLE
Feature-Request - Graph by label

### ISSUE THREAD

**Issue by Heiko-san** [SamAmco/track-and-graph#236]
Hi,
your app is really great!

However I feel it would be very useful, when creating a graph, that you could "filter" the values to those with a specific label. I mean some kind of "include a set of labels" functionality for the graph settings.

Let's say you keep track of your pulse for example. So you have measurements with resting pulse and those with pulse while doing sports. Now maybe you want to graph your average resting pulse. If your values have labels like "rest", "sport", ... You could do that by only graph the values with the label "rest".
However of course the app doesn't support such a filter at the moment.

Maybe it could be useful the other way round as well: That you can exclude a set of labels from graph.

**Comment by SamAmco** [comment]
Hi, I think this is another use case for functions which i'm planning on implementing at some point in future.

**Comment by Heiko-san** [comment]
Maybe, I just saw what is possible right now and the obvious solution seemed to be "filter by label" :)

**Comment by SamAmco** [comment]
Yeah it's another one of these things where i want to avoid complicating the graph interface as much as possible and extract that complexity for advanced users. There are all kinds of things you might want to do to your data before plotting it. Filtering by label is a good example.

**Comment by Heiko-san** [comment]
The more I think about it, the more I like the idea of having functions for that. :)

**Comment by SamAmco** [comment]
I will leave this open as I think it would be great to have a script for it on the [community page](https://samamco.github.io/track-and-graph/lua/community/). But fyi this is now totally do-able using [Lua graphs](https://samamco.github.io/track-and-graph/tutorial/graphs/lua-graphs/lua-graphs/)

**Comment by SamAmco** [comment]
Functions do this cleanly now. Closing.
